web/Services/PostcodeService.cs

```diff
@@ -0,0 +1,176 @@
+        try
+        {
+            var response = await _client.GetAsync(uri);
+            response.EnsureSuccessStatusCode();
+            responseBody = await response.Content.ReadAsStringAsync();
+            _logger.LogInformation("Successful API request to {Uri}", uri);
+        }
+        catch (Exception e)
+        {
+            _logger.LogWarning("Message :{Message} ", e.Message);
+        }
+
+        return responseBody;
+    }
+    
+    private PostcodeioResponseModel TurnResponseBodyToModel(string responseBody)
+    {
+        if (string.IsNullOrEmpty(responseBody))
+        {
+            return new PostcodeioResponseModel();
+        }
+        var postcodeResponse = JsonConvert.DeserializeObject<PostcodeioResponseModel>(responseBody);
+        return postcodeResponse!;
+    }
+
+    public async Task BulkAddPostcodesToDatabaseIfAbsent(List<string> postcodes)
+    {
+        var postcodesToLookup = new List<string>();
+        foreach (string postcode in postcodes)
+        {
+            if (!_dbContext.Postcodes.Any(p => p.Postcode == postcode) && !postcodesToLookup.Any(p => p == postcode))
+            {
+                postcodesToLookup.Add(postcode);
+            }
+
+            if (postcodesToLookup.Count() == 100)
+            {
+                await BulkGetLatLonForPostcodes(postcodesToLookup);
+                postcodesToLookup = new List<string>();
+            }
+        }
+
+        if (postcodesToLookup.Count() != 0)
+        {
+            await BulkGetLatLonForPostcodes(postcodesToLookup);
+        }
+    }
+
+    private async Task BulkGetLatLonForPostcodes(List<string> postcodesToLookup)
+    {
+        var bulkResponseBody = await MakeBulkApiRequestToPostcodeApi(postcodesToLookup);
+        var postcodeResponse = TurnBulkResponseBodyToModel(bulkResponseBody);
+
+        foreach (PostcodeioResponseModel response in postcodeResponse)
+        {
+            AddPostcodeToPostcodeDb(response);
+        }
+    }
+
+    private async Task<string> MakeBulkApiRequestToPostcodeApi(List<string> postcodes)
+    {
+        var postcodesJson = new { postcodes = postcodes.ToArray() };
+        var uri = "https://api.postcodes.io/postcodes/";
+        var bulkResponseBody = string.Empty;
+        _logger.LogInformation("Making API Request to {Uri}", uri);
+        try
+        {
+            var response = await _client.PostAsJsonAsync(uri, postcodesJson);
+            response.EnsureSuccessStatusCode();
+            bulkResponseBody = await response.Content.ReadAsStringAsync();
+            _logger.LogInformation("Successful API request to {Uri}", uri);
+        }
+        catch (Exception e)
+        {
+            _logger.LogWarning("Message :{Message} ", e.Message);
+        }
+
+        return bulkResponseBody;
+    }
+
+    private List<PostcodeioResponseModel> TurnBulkResponseBodyToModel(string bulkResponseBody)
+    {
+        var postcodeDbModelList = new List<PostcodeioResponseModel>();

```

Comment:
I dont think this variable name is accurate :)

Reply:
I'd have to agree with you there - I think the name was left-over from an earlier version of the code, but looking at it now it obviously doesn't make sense.